sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/MockEventProcessorClient.cs

```diff
@@ -0,0 +1,370 @@
+        /// <param name="clientOptions">The set of options to use for this processor.</param>
+        /// <param name="fakePartitionProcessing"><c>true</c> if <see cref="RunPartitionProcessingAsync" /> should be overridden; otherwise, <c>false</c>.</param>
+        /// <param name="numberOfPartitions">The amount of partitions the associated Event Hub has.</param>
+        ///
+        internal MockEventProcessorClient(PartitionManager storageManager,
+                                          string consumerGroup = "consumerGroup",
+                                          string fullyQualifiedNamespace = "somehost.com",
+                                          string eventHubName = "somehub",
+                                          Func<EventHubConnection> connectionFactory = default,
+                                          EventProcessorClientOptions clientOptions = default,
+                                          bool fakePartitionProcessing = true,
+                                          int numberOfPartitions = 3) : base(storageManager, consumerGroup, fullyQualifiedNamespace, eventHubName, connectionFactory, clientOptions)
+        {
+            StorageManager = storageManager;
+
+            var partitionIds = Enumerable
+                .Range(1, numberOfPartitions)
+                .Select(p => p.ToString())
+                .ToArray();
+
+            foreach (var partitionId in partitionIds)
+            {
+                EventPipeline[partitionId] = new ConcurrentQueue<EventData>();
+            }
+
+            FakeRunPartitionProcessingAsync = fakePartitionProcessing;
+
+            ProcessErrorAsync += eventArgs =>
+            {
+                Exception[] newException = new Exception[] { eventArgs.Exception };
+                ExceptionCalls.AddOrUpdate(
+                    eventArgs.PartitionId,
+                    newException,
+                    (partitionId, value) => value.Concat(newException).ToArray());
+
+                return Task.CompletedTask;
+            };
+
+            ProcessEventAsync += eventArgs =>
+            {
+                EventData[] newEvent = new EventData[] { eventArgs.Data };
+                ProcessEventCalls.AddOrUpdate(
+                    eventArgs.Partition.PartitionId,
+                    newEvent,
+                    (partitionId, value) => value.Concat(newEvent).ToArray());
+
+                return Task.CompletedTask;
+            };
+
+            PartitionInitializingAsync += eventArgs =>
+            {
+                InitializeCalls.AddOrUpdate(eventArgs.PartitionId, 1, (partitionId, value) => value + 1);
+
+                return Task.CompletedTask;
+            };
+
+            PartitionClosingAsync += eventArgs =>
+           {
+               CloseCalls.AddOrUpdate(eventArgs.PartitionId, 1, (partitionId, value) => value + 1);
+               StopReasons[eventArgs.PartitionId] = eventArgs.Reason;
+
+               return Task.CompletedTask;
+           };
```

Comment:
Final nit comment: it seems the indentation is 1 character off here. `PartitionClosingAsync += eventArgs =>` seems to be right, but the whole following block should be 1 position to the right.